File: ruby/lib/casilla_sorpresa.rb
module Civitas
    class CasillaSorpresa < Casilla
        def initialize(mazo)
            super("Sorpresa")
            @mazo=mazo
            @sorpresa=nil
        end
        attr_reader :mazo, :sorpresa

        def recibeJugador(actual, todos)
            if (jugadorCorrecto(actual, todos))
                informe(actual, todos)
                @sorpresa = @mazo.siguiente
                @sorpresa.aplicarAJugador(actual, todos)
        
            end
        end
    end
end

File: Civitas_gui/src/civitas/CasillaSorpresa.java
package civitas;
import java.util.ArrayList;

public class CasillaSorpresa extends Casilla {
    private Sorpresa sorpresa;
    private MazoSorpresas mazo;
    CasillaSorpresa(MazoSorpresas mazo, String nombre) {
        super(nombre);
        this.mazo = mazo;
    }
    
    @Override
    public String getTipo(){
        return "Sorpresa";
    }

    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            sorpresa = mazo.siguiente();
            sorpresa.aplicarAJugador(actual, todos);
        }
    }
}